src/PropelSqlExtractor.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+namespace RebelCode\Migrations;
+
+use Dhii\Exception\CreateInvalidArgumentExceptionCapableTrait;
+use Dhii\I18n\StringTranslatingTrait;
+use Dhii\Util\Normalization\NormalizeStringCapableTrait;
+use Dhii\Util\String\StringableInterface as Stringable;
+
+/**
+ * A Propel SQL extractor that stores the extracted SQL in files that can be used by RebelCode and ByJg migration
+ * libraries.
+ *
+ * @since [*next-version*]
+ */
+class PropelSqlExtractor extends AbstractFilePropelSqlExtractor implements PropelSqlExtractorInterface
+{
+    /*
+     * Filters extracted SQL queries to add SQL that simulates Propel migration.
+     *
+     * @since [*next-version*]
+     */
+    use MockPropelFilterQueryCapableTrait;
+
+    /*
+     * Provides string normalization capabilities.
+     *
+     * @since [*next-version*]
+     */
+    use NormalizeStringCapableTrait;
+
+    /*
+     * Provides functionality for creating invalid argument exception instances.
+     *
+     * @since [*next-version*]
+     */
+    use CreateInvalidArgumentExceptionCapableTrait;
+
+    /*
+     * Provides string translating capabilities.
+     *
+     * @since [*next-version*]
+     */
+    use StringTranslatingTrait;
+
+    /**
+     * The directory name for up migrations.
+     *
+     * @since [*next-version*]
+     */
+    const DIR_UP = 'up';
+
+    /**
+     * The directory name for down migrations.
+     *
+     * @since [*next-version*]
+     */
+    const DIR_DOWN = 'down';
+
+    /**
+     * The root directory.
+     *
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $rootDir;
+
+    /**
+     * Constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $rootDir The absolute path to the root directory where migration SQL files are stored.
+     */
+    public function __construct($rootDir)
+    {
+        $this->_setRootDir($rootDir);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function extract($migration)
+    {
+        return $this->_extract($migration);
+    }
+
+    /**
+     * Retrieves the root directory where migration SQL files are stored.
+     *
+     * @since [*next-version*]
+     *
+     * @return string The absolute path to the root directory.
+     */
+    protected function _getRootDir()
+    {
+        return $this->rootDir;
+    }
+
+    /**
+     * Sets the root directory where migration SQL files are stored.
+     *
+     * @since [*next-version*]
+     *
+     * @param string $rootDir The absolute path to the root directory.
+     */
+    protected function _setRootDir($rootDir)
+    {
+        $this->rootDir = $rootDir;
+    }
+
+    /**
+     * {@inheritdoc}
+     *
```

Comment:
I feel that this implementation deserves some explanation of how it deduces the path, even if only a shallow overview.